Meta/generate_encoding_indexes.py

```diff
@@ -0,0 +1,298 @@
+        content += generate_table(name, table) + "\n\n"
+
+    content += "}\n"
+
+    with open(output_path, "w") as f:
+        f.write(content)
+
+
+def generate_table_accessor(name: str, table: LookupTable) -> str:
+    if table.first_pointer > 0:
+        return f"""
+Optional<u32> index_{name}_code_point(u32 pointer)
+{{
+    if (pointer < s_{name}_index_first_pointer || pointer - s_{name}_index_first_pointer >= s_{name}_index.size())
+        return {{}};
+    auto value = s_{name}_index[pointer - s_{name}_index_first_pointer];
+    if (value == 0xfffd)
+        return {{}};
+    return value;
+}}
+"""
+    else:
+        return f"""
+Optional<u32> index_{name}_code_point(u32 pointer)
+{{
+    if (pointer >= s_{name}_index.size())
+        return {{}};
+    auto value = s_{name}_index[pointer];
+    if (value == 0xfffd)
+        return {{}};
+    return value;
+}}
+"""
+
+
+def generate_inverse_table_accessor(name: str, table: LookupTable) -> str:
+    if table.first_pointer > 0:
+        return f"""
+Optional<u32> code_point_{name}_index(u32 code_point)
+{{
+    for (u32 i = 0; i < s_{name}_index.size(); ++i) {{
+        if (s_{name}_index[i] == code_point) {{
+            return s_{name}_index_first_pointer + i;
+        }}
+    }}
+    return {{}};
+}}
+"""
+    else:
+        return f"""
+Optional<u32> code_point_{name}_index(u32 code_point)
+{{
+    for (u32 i = 0; i < s_{name}_index.size(); ++i) {{
+        if (s_{name}_index[i] == code_point) {{
+            return i;
+        }}
+    }}
+    return {{}};
+}}
+"""
+
+
+def generate_implementation_file(tables: LookupTables, output_path: Path) -> None:
+    content = """
+#include <LibTextCodec/LookupTables.h>
+
+namespace TextCodec {
+"""
+
+    for name, table in tables.indexes.items():
+        if table.generate_accessor:
+            content += generate_table_accessor(name, table)
+        if table.generate_inverse_accessor:
+            content += generate_inverse_table_accessor(name, table)
+
+    content += "\n}\n"
+
+    with open(output_path, "w") as f:
+        f.write(content)
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Generate text codec lookup tables", add_help=False)
+    parser.add_argument("-h", required=True)
+    parser.add_argument("-c", required=True)
+    parser.add_argument("-j", required=True)
+
+    args = parser.parse_args()
+
+    with open(args.j, "r") as f:
+        data = json.load(f)
+
+    gb18030_table = prepare_table(data["gb18030"], GenerateAccessor.YES)
+
+    gb18030_updates = {
+        7182: 0xFE10,
+        7183: 0xFE12,
+        7184: 0xFE11,
+        7185: 0xFE13,
+        7186: 0xFE14,
+        7187: 0xFE15,
+        7188: 0xFE16,
+        7201: 0xFE17,
+        7202: 0xFE18,
+        7208: 0xFE19,
+        23775: 0x9FB4,
+        23783: 0x9FB5,
+        23788: 0x9FB6,
+        23789: 0x9FB7,
+        23795: 0x9FB8,
+        23812: 0x9FB9,
+        23829: 0x9FBA,
+        23845: 0x9FBB,
+    }
```

Comment:
when I mentioned that these updates were already merged into the encoding spec, I meant "a future PR should update the json to match the current spec and remove this manual dict update", rather than "comment and spec ref not needed".